extensions/wikia/AdEngine/js/template/interstitial.js

```diff
@@ -0,0 +1,60 @@
+/*global define*/
+define('ext.wikia.adEngine.template.interstitial', [
+	'wikia.document',
+	'wikia.log',
+	'wikia.ui.factory',
+	'wikia.window'
+], function (doc, log, uiFactory, win) {
+	'use strict';
+
+	var logGroup = 'ext.wikia.adEngine.template.interstitial',
+		modalId = 'ext-wikia-adEngine-template-interstitial',
+		modalPadding = 20,
+		pollingInterval = 1000; // 1s
+
+	function show(params) {
+		log(['show', params], 'debug', logGroup);
+
+		var iframe = doc.getElementById(params.slot).querySelector('iframe'),
+			iframeBody = iframe.contentWindow.document.body,
+			selector = params.elem;
+
+		function onFound(element) {
+			var modalConfig = {
+					vars: {
+						id: modalId,
+						size: params.size || 'medium',
+						content: element.outerHTML,
+						title: 'Advertisement',
+						closeText: 'Close',
+						buttons: []
+					}
+				};
+
+			uiFactory.init('modal').then(function (uiModal) {
+				uiModal.createComponent(modalConfig, function (modal) {
+					var width = params.width || modal.$content[0].querySelector(selector).clientWidth;
+					modal.$element.width(width + 2 * modalPadding);
+					modal.show();
+				});
+			});
+		}
+
+		function waitForElement() {
```

Comment:
Have you considered using [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) where available instead of polling? TBH, I haven't had a chance to use it myself but it might be a performance improvement. We'd still need the polling for IE<11.